kafkakewl-metrics/src/test/scala/com/mwam/kafkakewl/metrics/services/KafkaConsumerMetricsCalcTest.scala

```diff
@@ -0,0 +1,272 @@
+      cmc: ZStream[Any, Nothing, KafkaConsumerGroupMetricChanges],
+      topicQueue: Queue[KafkaTopicPartitionInfoChanges]
+  ): UIO[KafkaConsumerGroupMetricChanges] = for {
+    _ <- topicQueue.offer(
+      KafkaTopicPartitionInfoChanges(
+        Map.newBuilder
+          .addOne(KafkaTopicPartition("testTopic", 0) -> KafkaTopicPartitionInfo(20, 3, Instant.MAX))
+          .result(),
+        Set.empty
+      )
+    )
+    result <- cmc.take(1).runCollect
+  } yield result(0)
+
+  def deleteTestConsumerGroup(
+      cmc: ZStream[Any, Nothing, KafkaConsumerGroupMetricChanges],
+      consumerQueue: Queue[KafkaConsumerGroupOffsets]
+  ): UIO[KafkaConsumerGroupMetricChanges] = for {
+    _ <- consumerQueue
+      .offer(
+        Map.newBuilder
+          .addOne(ConsumerGroupTopicPartition("testGroup", "testTopic", 0) -> None)
+          .result()
+      )
+    result <- cmc.take(1).runCollect
+  } yield result(0)
+
+  def deleteTestTopic(
+      cmc: ZStream[Any, Nothing, KafkaConsumerGroupMetricChanges],
+      topicQueue: Queue[KafkaTopicPartitionInfoChanges]
+  ): UIO[KafkaConsumerGroupMetricChanges] = for {
+    _ <- topicQueue
+      .offer(
+        KafkaTopicPartitionInfoChanges(
+          Map.empty,
+          Set(KafkaTopicPartition("testTopic", 0))
+        )
+      )
+    result <- cmc.take(1).runCollect
+  } yield result(0)
+
+  def kafkaConsumerGroupMetricChangesWithoutTopic: KafkaConsumerGroupMetricChanges = KafkaConsumerGroupMetricChanges(
+    Map.newBuilder
+      .addOne(
+        ConsumerGroupTopicPartition("testGroup", "testTopic", 0) -> KafkaConsumerGroupMetrics(
+          ConsumerGroupStatus.Ok,
+          None,
+          Some(KafkaConsumerGroupOffset(123, "", OffsetDateTime.MAX)),
+          None,
+          Some(0.0)
+        )
+      )
+      .result(),
+    Set.empty
+  )
+
+  def kafkaConsumerGroupMetricChangesWithTopic: KafkaConsumerGroupMetricChanges = KafkaConsumerGroupMetricChanges(
+    Map.newBuilder
+      .addOne(
+        ConsumerGroupTopicPartition("testGroup", "testTopic", 0) -> KafkaConsumerGroupMetrics(
+          ConsumerGroupStatus.Ok,
+          Some(KafkaTopicPartitionInfo(20, 3, Instant.MAX)),
+          Some(KafkaConsumerGroupOffset(123, "", OffsetDateTime.MAX)),
+          Some(0),
+          Some(0.0)
+        )
+      )
+      .result(),
+    Set.empty
+  )
+
+  def kafkaConsumerGroupMetricChangesRemoved: KafkaConsumerGroupMetricChanges = KafkaConsumerGroupMetricChanges(
+    Map.empty,
+    Set(ConsumerGroupTopicPartition("testGroup", "testTopic", 0))
+  )
+
+  def spec: Spec[Any, Nothing] = suite("KafkaConsumerMetricsCalc Tests")(
+    test("Adding a kafka consumer group starts adding consumer group metrics") {
+      for {
+        start <- getQueuesAndMetricsCalc
+        (_, consumerQueue, cmc) = start
+        firstResult <- pushTestConsumerGroup(cmc, consumerQueue)
+
+      } yield assertTrue(
+        firstResult == KafkaConsumerGroupMetricChanges(
+          Map.newBuilder
```

Comment:
nitpick, but you can use the `Map(key1 -> value1, key2 -> value2)` syntax too which is a bit more concise and readable.

Reply:
Good spot. I have updated this in all occurrences in the test code